### ISSUE TITLE
[BUG] Old checkbox state in result cards produces blank output file

### ISSUE THREAD

**Issue by jarmoza** [neurobagel/old-query-tool#124]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Expected Behavior

Initiating a successive query should produce result cards that are all unchecked, and additionally, downloaded results should have the appropriate subject- or dataset-level tsv data.

### Current Behavior

1. After initiating a query (i.e. querying all the datasets without filters) and selecting a few of the dataset results by clicking on their checkboxes, successive queries afterwards should have those checked checkbox states be cleared to unchecked.
2. When downloading results from such successive queries, proper results should be in the downloaded tsv file.

### How to reproduce

1. Query all
2. Select a few datasets in the results by clicking on their checkboxes
3. Alter the query filter(s)
4. Query again
5. Note that checkboxes in the new returned results remain checked
6. Attempt to download subject- or dataset-level datasets from the results
7. Note that the output file is blank except for the header row

**Comment by rmanaem** [comment]
Discovered:
- keys in `v-for` need to be unique across session
- `checked` attribute of the `input` element sets the initial state of the checkbox
- If the `checked` attribute is used without vue interference i.e. ```<input checked="something" />``` the initial state is going to be true
- If the `checked` attribute is `v-binded` it can reflect the truthy/falsy value of the statement passed to it i.e. 
- ```<input :checked="false" />``` will set the initial state to be false
- If the `checked attribute is `v-binded` to a prop, aside from using the prop value for the initial state, it will reflect the value of the prop unless the prop is changed i.e. ```<input :checked="selectAll" />``` aside from the initial state of the checkbox, the `selectAll` only affects the state checkbox if its changed.

To be discovered
- Uncheck checkboxes consistently
- WIpe `downloads` array

**Comment by surchs** [comment]
~~I'm thinking if we could force vue to rerender here simply by using vue-model in addition to the event listener. Right now the checkbox status is not part of the state of the result card. But if we model it it would be~~

edit: I think I was half-right here: The problem with the checkboxes is related to an imcomplete modeling of state. When I create a ResultsCard, I tell it whether it should be "checked=true" or "checked=false" by passing `selectAll` as a prop. If `selectAll` changes later, this change will propagate down, and the checkboxes will update accordingly. Neat!

But that's only half of it. When a user clicks on the checkbox, we listen to the @change event. But we don't update the prop that controls the "checked-ness" status. This only looks as if it works because clicking on a checkbox item causes the browser to make the checkbox look "checked". What we really need to do is take full control of the visual state of the checkbox by also updating the prop we're passing in response to a `@change` event, not only when the `selectAll` variable changes.

In other words: it is the responsibility of `ResultContainer` to 
- handle the state for each `ResultCard` and
- to listen to emitted events from all `ResultCard`s and update the state accordingly and
- to pass down updates to the state to the respective `ResultCard`s
